### DISCUSSION TITLE
auto update "last_modified"

### DISCUSSION THREAD

**Discussion by dtau00** [appy-one/acebase#84] [Q&A]
mongodb has a feature where it auto updates a "last_modified" field whenever a record is updated.  is there a similar feature here?  if not, its there a preferred way to handle this?  Getting the current date from the client will be inaccurate.  I could watch for mutations on the server and update the record afterwards, but i think that'll cause every modification to trigger 2 update events (or even an infinite loop).  I could get the server date before updating, but that could have problems too (if offline or temporarily not connected)

**Comment by appy-one** [comment]
You can prevent a loop by providing a context in your update:

```js
db.ref('my_collection').on('notify_child_changed', ref => {
   const context = ref.context();
   if (context && context.lastModifiedEvent) {
      return; // Prevent recursive updates
   }
   // Update last_modified
   ref.context({ lastModifiedEvent: true }).update({ last_modified: Date.now() });
});
```

Also see [Get triggering context of events](https://github.com/appy-one/acebase#get-triggering-context-of-events)
